pkg/destroy/openstack/openstack.go

```diff
@@ -624,15 +624,14 @@ func deleteCustomRouterInterfaces(opts *clientconfig.ClientOpts, filter Filter,
 		return false, nil
 	}
 
-	// Find the machines Network by the master machines Ports.
+	// Find the ports that should be removed from the router.
 	// Any worker nodes ports, including the ones used for additionalNetworks,
 	// are tagged with cluster-api-provider-openstack and should be excluded
 	// to ensure only the primary Network ports are filtered.
 	tags := filterTags(filter)
 	listOpts := ports.ListOpts{
-		Tags:        strings.Join(tags, ","),
-		DeviceOwner: "compute:nova",
-		NotTags:     "cluster-api-provider-openstack",
+		Tags:    strings.Join(tags, ","),
+		NotTags: "cluster-api-provider-openstack",
```

Comment:
This is needed to avoid handling a possible secondary Network on the Worker Nodes, so we fetch only master Ports and grab Network from there